evm_loader/emulator/src/solana_backend.rs

```diff
@@ -0,0 +1,382 @@
+        if accounts.get(&address).is_none() {
+            eprintln!("Not found account for {}", &address.to_string());
+
+            let (solana_address, _) = Pubkey::find_program_address(&[&address.to_fixed_bytes()], &self.program_id);
+            
+            match self.rpc_client.get_account(&solana_address) {
+                Ok(acc) => {
+                    eprintln!("Account found");                        
+                    eprintln!("Account data len {}", acc.data.len());
+                    eprintln!("Account owner {}", acc.owner.to_string());
+                   
+                    accounts.insert(address, SolidityAccount::new(acc.data, acc.lamports).unwrap());
+
+                    true
+                },
+                Err(_) => {
+                    eprintln!("Account not found {}", &address.to_string());
+
+                    new_accounts.insert(address);
+
+                    false
+                }
+            }
+        } else {
+            true
+        }
+    }
+
+    pub fn apply<A, I, L>(&mut self, values: A, _logs: L, _delete_empty: bool)
+            where
+                A: IntoIterator<Item=Apply<I>>,
+                I: IntoIterator<Item=(H256, H256)>,
+                L: IntoIterator<Item=Log>,
+    {             
+        let mut accounts = self.accounts.borrow_mut(); 
+
+
+        for apply in values {
+            match apply {
+                Apply::Modify {address, basic, code: _, storage: _, reset_storage} => {
+                    match accounts.get_mut(&address) {
+                        Some(acc) => {
+                            *acc.updated.borrow_mut() = true;
+                        },
+                        None => {
+                            eprintln!("Account not found {}", &address.to_string());
+                        },
+                    }
+                    eprintln!("Modify: {} {} {} {}", &address.to_string(), &basic.nonce.as_u64(), &basic.balance.as_u64(), &reset_storage.to_string());
+                },
+                Apply::Delete {address: addr} => {
+                    eprintln!("Delete: {}", addr.to_string());
+                },
+            }
+        };
+    }
+
+    pub fn get_used_accounts(&self)
+    {          
+        let mut arr = Vec::new();    
+        
+        eprint!("[");
+        let accounts = self.accounts.borrow();
+        for (address, acc) in accounts.iter() {
+            arr.push(AccountJSON{address: "0x".to_string() + &hex::encode(&address.to_fixed_bytes()), readonly: !&acc.updated, new: false});
+            eprint!("{{\"address\":\"0x{}\",\"write\":\"{}\"}},", &hex::encode(&address.to_fixed_bytes()), &acc.updated.to_string());
+        }
+        let new_accounts = self.new_accounts.borrow(); 
+        for address in new_accounts.iter() {
+            arr.push(AccountJSON{address: "0x".to_string() + &hex::encode(&address.to_fixed_bytes()), readonly: false, new: true});
```

Comment:
It's better to return `writable` flag instead of `readonly` (AccountMeta in Solana has `is_writable` properties)